File: TelerikUIExamples/CategoricalAxis.m
//
//  CategoricalAxis.m
//  TelerikUIExamples
//
//  Copyright (c) 2013 Telerik. All rights reserved.
//

#import "CategoricalAxis.h"
#import <TelerikUI/TelerikUI.h>

@implementation CategoricalAxis
{
    TKChart *_chart;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    _chart = [[TKChart alloc] initWithFrame:[self exampleBounds]];
    _chart.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    [self.view addSubview:_chart];
    
    NSMutableArray *array = [[NSMutableArray alloc] init];
    NSArray *categories = @[ @"Apple", @"Google", @"Microsoft", @"Samsung" ];
    for (int i = 0; i<categories.count; i++) {
        [array addObject:[[TKChartDataPoint alloc] initWithX:categories[i] Y:@(arc4random() % 100)]];
    }
    TKChartColumnSeries *series = [[TKChartColumnSeries alloc] initWithItems:array];
    series.selectionMode = TKChartSeriesSelectionModeSeries;
    
    TKChartCategoryAxis *xAxis = [[TKChartCategoryAxis alloc] initWithCategories:categories];
    xAxis.position = TKChartAxisPositionBottom;
    xAxis.plotMode = TKChartAxisPlotModeBetweenTicks;
    _chart.xAxis = xAxis;
    
    TKChartNumericAxis *yAxis = [[TKChartNumericAxis alloc] initWithMinimum:@(0) andMaximum:@(100)];
    yAxis.position = TKChartAxisPositionLeft;
    _chart.yAxis = yAxis;
    
    [_chart addSeries:series];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: TelerikUIExamplesInCSharp/Examples/Chart/AxisTypes/CategoricalAxis.cs
using System;
using System.Collections.Generic;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	public class CategoricalAxis: ExampleViewController
	{
		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();

			TKChart chart = new TKChart (this.ExampleBounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (chart);

			Random r = new Random ();
			List<TKChartDataPoint> list = new List<TKChartDataPoint> ();
			string[] categories = new []{"Apple", "Google", "Microsoft", "Samsung"};
			for (int i = 0; i < categories.Length; i++) {
				list.Add(new TKChartDataPoint(new NSString(categories[i]), new NSNumber(r.Next() % 100))); 
			}

			TKChartColumnSeries series = new TKChartColumnSeries (list.ToArray());
			series.SelectionMode = TKChartSeriesSelectionMode.Series;

			TKChartCategoryAxis xAxis = new TKChartCategoryAxis ();
			xAxis.Position = TKChartAxisPosition.Bottom;
			xAxis.PlotMode = TKChartAxisPlotMode.BetweenTicks;
			series.XAxis = xAxis;

			TKChartNumericAxis yAxis = new TKChartNumericAxis (new NSNumber (0), new NSNumber (100));
			yAxis.Position = TKChartAxisPosition.Left;
			chart.YAxis = yAxis;

			chart.AddSeries (series);
		}
	}
}

